### DISCUSSION TITLE
Feature request: Add the ability to create PRs from a tag to a branch (production < v1.1.1)

### DISCUSSION THREAD

**Discussion by NMFR** [community/community#23870] [New to GitHub]
<p>Hello,</p>
<p>I kind of use PRs to validate deploys to environments:</p>
<p>So after i create a new release, via github releases, i get a tag, lets say <strong>v1.1.1</strong>.</p>
<p>If i want to deploy <strong>v1.1.1</strong> to the <strong>production</strong> environment, i open a PR from a branch that is at <strong>v1.1.1</strong> to the  <strong>production</strong> branch. </p>
<p>The team than decides if it approves the PR and when merging the CI/CD kicks in when a new commit is pushed to  <strong>production</strong>.</p>
<p>It would be cool if i could open the PRs directly from the tag <strong>v1.1.1</strong> to the  <strong>production</strong>  branch, instead of having a branch pointing to <strong>v1.1.1</strong> just for the PR.</p>
<p>Something like  <strong>production </strong> &lt;  <strong>v1.1.1</strong> : <a href="https://github.com/NMFR/my-repo/compare/production...v1.1.1?expand=1" rel="nofollow noopener">https://github.com/NMFR/my-repo/compare/production…v1.1.1?expand=1</a></p>
<p>Thanks in advance.</p>

**Comment by BelHFL** [comment, marked as answer]
<p>Hi <a class="mention" href="/nmfr">@nmfr</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Cheers!</p>

**Reply by guruprasadsingirikonda** [reply]
Hi Team,

When will this feature be available? I've GitHub Enterprise Servers set up in a data center. 

In GHES there is no import option available in the console, Please add this feature too.

**Reply by ghost** [reply]
Hi @BelHFL: Can you please post any update on this feature request?

**Reply by hwmrocker** [reply]
I would also be interested in this feature

**Reply by evktalo** [reply]
I right now am working on a contribution where I would be happy to have this feature!

**Reply by samuel-andres** [reply]
please do it

**Comment by seveas** [comment]
<p>Whilw your feature request makes perfect sense in your case, I think it is not as intuitive in general: pull requests are not static pointers to a commit, but you can push new commits to a PR, force-push entirely new contents to it and do all things that in the world of git are meant for branches and not tags.</p>
<p>What is your normal branching model for your major/minor/feature releases? Maybe there’s a middle ground to be found that works.</p>

**Comment by IgnusG** [comment]
<p>I’ve also just stumbled upon this issue, exactly the same workflow:<br>
I have a master branch, a pre-release branch and a release branch.</p>
<p>I can create PR’s from master to pre-release where I can edit changes and finally issue a command to create a version bump, tag and merge to the pre-release. This is a beta branch which receives updates often.</p>
<p>Once a month I want to merge a specific version from the developer branch into the release branch. This should not be the HEAD of pre-release since there might be changes on it which have not been tested thoroughly enough. It would be awesome if I could create a PR from a specific version tag (which are on the pre-release) to release and merge if everything checks out.</p>
<p>In this case, it doesn’t make sense to allow changes to the PR. The version should be immutable so if I want to change what is merged to public release I would close such a PR and create a new one with HEAD from another version tag.</p>
<blockquote>
<p>The automation for these is done using GitHub Actions which is the reason I chose PR’s as a release tool</p>
</blockquote>

**Comment by francisu** [comment]
<p>Another use case for this is when I fork a repo, and I want to make a PR against a previous release for which there is no branch in the original repo, only a tag. And since I don’t control the original repo, I can’t make a branch.</p>

**Comment by JameelKhan9** [comment]
Facing the same use case as OP.

**Comment by baodoan-tradedoubler** [comment]
Yes, I also need this feature from github.

**Comment by sanamhub** [comment]
I too have an use case for this and upvoted as well!

> Use case

I've not created production branch previously pointing to particular tag (only tracked via tag) and I want to create new branch and open PR from that tag to my new branch

**Comment by dimpase** [comment]
Another important issue to resolve here is doing PRs for GitHub Actions. Versioning of the latter is done via tags, and without ability to change tags in PRs, PRs on Actions afer merging need manual adjustment of tags in master.

**Comment by TheaKathrine** [comment]
This feature would be great, along with some tag protection rules

**Comment by san1t1** [comment]
I'd be keen to see this feature too.

**Comment by owais-naim** [comment]
Today I faced this same situation. It would be great if Github had this feature.

**Comment by sanamhub** [comment]
@github

**Comment by kumaarn** [comment]
waiting for this feature

**Comment by eusoumabel** [comment]
This feature is needed!!

**Comment by joshuapetersse** [comment]
I'll upvote this one, too.  I just ran into a case where I need this, and although you can select a tag while creating a PR, the actual creation of it is blocked once you do.

**Comment by simon-brooke** [comment]
I think this feature is most necessary and urgent.

I recently created a pull request to a parent project, and then continued development in my own repository preparing for the next major release; all my pushes got appended to the pull request, polluting it and rendering it useless.

**Comment by fguillen** [comment]
+1

**Comment by lorenjohnson** [comment]
+1

**Comment by Vyko** [comment]
+1

**Comment by sanamhub** [comment]
+1

**Comment by dsamson-pluscare** [comment]
+1

**Comment by florent-foxtel** [comment]
+1

**Comment by ravibagri4** [comment]
+1

**Comment by oshmytov** [comment]
I need this feature too for the reasons above.

**Comment by BacLuc** [comment]
i would like this feature too.
We currently deploy from branches and have the 4 eye principle via PR review.

**Comment by samueldc** [comment]
Creating pull requests from tagged commits would also be great in my workflow too. 

+1

**Comment by gabriele-lagana-st** [comment]
Having this feature will avoid me to create a new branch from a tag and then delete it only to create the PR.
+1

**Comment by sarkoudelis** [comment]
+3
